### PR TITLE
Allow partial translate file reads.

### LINKED ISSUE #1860: translate store buffer exceeded max size: pilosa: translate read target is undersized
### Description
This error occurs when an import into an index or field using key translation has keys which exceed the maximum size of a translation replication entry (set to 4MB at time of writing).

Currently the pilosa.translateFileReader is limited to reading an entire LogEntry at once. The LogEntry is created by a single call to (e.g.) `TranslateColumnsToUint64` which can contain an arbitrary number of keys to be translated (and I think each key can be of arbitrary size). 

There are a few possible avenues to fixing this. Breaking up LogEntries automatically is one, but it would probably be better to improve the translateFileReader.Read implementation to be more flexible (not need to read an entire entry at once), and then basically ioutil.Copy that directly to the ResponseWriter in handler.GetTranslateData.

### Success criteria (What criteria will consider this ticket closeable?)

Large import requests should not be able to cause errors with key translation replication just because there is a maximum read buffer size.

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This commit fixes an issue where translation `LogEntry` must be read in its entirety, however, large entries can exceed the buffer size. This has been changed so that partial entries reads are allowed.

The `LogEntry.ReadFrom()` may still generate large byte slices during reads of large individual fields or keys.

Fixes #1860

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.
- [x] Make sure PR title conforms to convention in CHANGELOG.md.
- [x] Add appropriate changelog label to PR (if applicable).

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.
- [ ] Make sure PR title conforms to convention in CHANGELOG.md.
- [ ] Make sure PR is tagged with appropriate changelog label.

### CODE REVIEW

File: translate.go [hunk 1]
```diff
@@ -1137,20 +1111,14 @@ func (r *translateFileReader) read(p []byte) (n int, err error) {
...
-		// size of p and try again.
-		return 0, ErrTranslateReadTargetUndersized
-	} else if int64(len(p)) > max {
+	if max := sz - r.offset; int64(len(p)) > max {
 		// Shorten buffer to maximum read size.
 		p = p[:max]
 	}
 
 	// Read data from file at offset.
 	// Limit the number of bytes read to only whole entries.
```
Comment (jaffee) [inline translate.go]: I guess this comment needs to change now?

Comment (jaffee) [reply translate.go]: ...and potentially the help on this method since we might not get an integer number of valid entries.

**Comment by benbjohnson** [comment]
@jaffee Yes, `binary.Read()` will wait until it receives all the bytes necessary. I feel like there was a reason I went with this approach originally but honestly I can't remember now. I think I wanted readers to always start fetching from the start of an entry but that's really not necessary.

**Comment by benbjohnson** [comment]
@jaffee Can you +1 this updated branch and I'll merge it in.

**Pull request merged**